web/src/app/stuff-list/stuff-list.component.ts

```diff
@@ -30,7 +32,7 @@ export class StuffListComponent implements OnInit {
       let stuff = this.stuffList[stuffIndex];
       console.log(stuff);
       if (stuff['id'] === id) {
-        this.stuffList.splice(stuffIndex, 1);
+        this.stuffList.splice(+stuffIndex, 1);
```

Comment:
What does this syntax mean? :)

Reply:
The `stuffIndex` variable here is of type string, so we need to convert it to number. Actually, I have no idea - why do we use `for...in` loop for an array instead of a common way of looping with `for(let i=0; i<this.stuffList; i++)`. There would be no need to convert anything - we'd just use `i` counter. It works, I checked.

Reply:
Sure, we can replace an implementation. But condition should be i<this.stuffList.**lenght** in this case.
But I still don't know why do you think stuffIndex can be a string if we have stuffList as an array.

Reply:
Of course it's `i<this.stuffList.lenght`, my bad )
stuffIndex has type string, i suppose `for...in` loop is the reason behind it since it's used to loop through objects (ex. `for key in object`) and doesn't go well with arrays. I can implement it through standard for-loop, quite neat.

Reply:
feel free to do it ;)